crates/iceberg-catalog/src/config.rs

```diff
@@ -73,6 +73,12 @@ pub struct DynAppConfig {
     pub nats_address: Option<Url>,
     #[clap(env = "ICEBERG_REST__NATS_TOPIC")]
     pub nats_topic: Option<String>,
+
+    // ------------- AUTHORIZATION -------------
+    #[clap(env = "ICEBERG_REST__OAUTH_PROVIDER_URI")]
+    pub openid_provider_uri: Option<Url>,
```

Comment:
Rename to oauth2?

Reply:
+Add to docs

Reply:
added to docs, decided to stick to OPENID_PROVIDER_URI since we are actually relying on openid endpoints like `.well-known/openid-configuration`.